packages/core/src/tools/utils.ts

```diff
@@ -564,3 +564,23 @@ export function combine(...sources: any[]): unknown {
 }
 
 export type TimeoutId = ReturnType<typeof setTimeout>
+
+export function requestIdleCallback(callback: () => void, opts?: { timeout?: number }) {
```

Comment:
I think we could still let this function here as it could be valuable at some point in other packages but let me know.